Extension/src/LanguageServer/client.ts

```diff
@@ -868,6 +877,87 @@ export class DefaultClient implements Client {
         return workspaceFolder ? workspaceFolder.name : "untitled";
     }
 
+    public updateClientConfigurations(): void {
+        clients.forEach(client => {
+            if (client instanceof DefaultClient) {
+                const defaultClient: DefaultClient = <DefaultClient>client;
+                defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                defaultClient.configuration.handleConfigurationChange();
+            }
+        });
+    }
+
+    public async handleCompilerQuickPick(): Promise<void> {
+        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
+        const settings: OtherSettings = new OtherSettings();
+        const paths: string[] = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
+        paths.push(localize("selectAnotherCompiler.string", "Select another compiler on my machine"));
+        paths.push(localize("installCompiler.string", "Help me install a compiler"));
+        paths.push(localize("noConfig.string", "Do not configure a compiler (not recommended)"));
+        const index: number = await ui.showSelectCompiler(paths);
+        if (index === paths.length - 1) {
+            paths[index] = "";
+        } else if (index === paths.length - 2) {
+            switch (os.platform()) {
+                case 'win32':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217614");
+                    return;
+                case 'darwin':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217706");
+                    return;
+                case 'linux':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217615");
+                    return;
+                default:
+                    throw new Error(localize("unexpected.os", "Unexpected OS"));
+            }
+        } else if (index === paths.length - 3) {
+            const result: vscode.Uri[] | undefined = await vscode.window.showOpenDialog();
+            if (result !== undefined && result.length > 0) {
+                util.addTrustedCompiler(result[0].fsPath);
+                settings.defaultCompiler = result[0].fsPath;
+                compilerDefaults = await this.requestCompiler(compilerPath.Value);
+                this.updateClientConfigurations();
+                return;
+            }
+        } else {
+            util.addTrustedCompiler(paths[index]);
```

Comment:
I'm seeing `null` get added a lot, but I'm not sure how that happened. When the user hits "escape" the index is -1, so the result is `undefined` which also gets added. I also see the empty string get added, not sure how. It seems like some checking should be added to prevent this persisted compiler list from having invalid data.